accepted/00102-upgrade-to-salt-3008.md

```diff
@@ -0,0 +1,199 @@
+
+At this point in time, only few modules have been already migrated to Salt Extensions, a lot of other modules are still waiting at [community-extensions-holding repository](https://github.com/salt-extensions/community-extensions-holding) to be migrated.
+
+The purpose of this RFC is to define:
+- Salt and Salt Extensions packaging strategy.
+- How to customize the Salt package with more extensions.
+- New dependencies required for Salt.
+- How to release Salt 3008.x to SLE.
+- How to maintain Salt 3008.x going forward.
+
+# Detailed design
+[design]: #detailed-design
+
+## Packaging
+
+Due the great module purge, we need to rethink the way we package Salt, as lot of the modules are been dropped and Salt Extensions are stepping in now.
+
+As upstream Salt project wants to drops the support from their side in favor of [community driven support](https://github.com/salt-extensions#what-are-salt-extensions) and lifecycle for the different Salt Extensions, this raises the question about which of them do we want to include and support.
+
+At the time of writting this RFC, most of the modules dropped from the Salt core are not yet migrated to Salt Extensions, therefore nobody from the community has claimed their ownership yet.
+
+### Builtin Salt Extensions
+
+There are a couple of Salt Extensions that we want to have integrated by default in our main `python311-salt` package. These are:
+- zypperpkg
+- dpkg
+- transactional_update
+- openscap
+- docker / dockerng
+- ...
+
+These are the minimum required extensions to be able to do basic operations in SUSE/openSUSE distributions and also allow basic operations on the context of SUSE Manager clients. We must maintain and support these builtin Salt Extensions.
+
+This list could be reduced even more by just providing the `zypperpkg` and `transactional_update` modules and then providing the rest via `/usr/share/susemanager/salt/[_modules,_states,...]` in the Uyuni / SUSE Manager server.
+
+Builtin Salt Extensions will be integrated into our main "openSUSE/salt" GitHub repo codebase for 3008.x, and provided as content of the main `python311-salt` package.
+
+#### An example of the Git repo:
+```
+salt
+|-- builin-extensions
+|   |-- saltext-zypperpkg
+|   |-- saltext-dpkg
+|   |-- ...
+...
+```
+
+The idea would be to include the releases for the different builtin extension directly inside our GH repo, as it was previously done with "tornado". We should take this oportunity also to take the ownership, migrate and create the official `saltext-zypperpkg` extension and probably some others.
+
+These builtin extensions will be then placed in `/usr/lib/python3.11/site-packages/saltext/` when installing `python311-salt` so they are available for Salt.
+
+**NOTE:** The usage of `git submodule` would make it a bit tricky when it comes to generate patches for the single unified Salt codebase, as apparentely you cannot get an unified diff for the main repo + submodules.
```

Comment:
Why do need to generate patches?

Reply:
I slightly modified the text, as generating patches would depend on the maintaining workflow we use.